src/api/cpp/seq.cpp

```diff
@@ -71,7 +71,8 @@ seq::seq(double begin, double end, double step): m_gfor(false)
     }
     if (end >= 0 && begin >= 0 && signbit(end-begin) != signbit(step))
         AF_THROW_ERR("Sequence is invalid", AF_ERR_ARG);
-        //AF_THROW("step must match direction of sequence");
+    if (begin>0 && end<0 && step>0)
```

Comment:
1) Shouldn't we also catch `af::seq(0, -15)`? i.e., `begin >= 0` instead of `begin > 0`.
2) Shouldn't we also catch `af::seq(-3, -15)`? i.e., drop `begin > 0` whatsoever?
2) This leads to the fact that negative indices are currently somehow special:
```
    af::seq(-2, -1, 1);   // passes, {-2, -1} when printed
    af::seq(-2, -1, -1);  // passes, crash when printed
    af::seq(2, 1, 1);     // does not pass
    af::seq(2, 1, -1);    // passes, {2, 1} when printed
```
Is this intentional? My question is, cannot we simply solve all the situations by removing `end >= 0 && begin >= 0` from the first condition or does it have a hidden purpose?